Doc/library/dis.rst

```diff
@@ -42,6 +42,16 @@ interpreter.
       bytecode to specialize it for different runtime conditions. The
       adaptive bytecode can be shown by passing ``adaptive=True``.
 
+   .. versionchanged:: 3.12
+      For instruction leading to a jump, a jump by 0 instruction should always
```

Comment:
What is an "instruction leading to a jump"?

Reply:
I agree it is a bit convoluted. I am happy to use a different phrasing. Initially I went for this because of opcode like `FOR_ITER` that do not jump often.

Reply:
I don't understand what you mean, so I can't offer how to reword it.

Reply:
Would the following re-wording read better ?

When the offset of a jump instruction is 0, the jump should always lead to the instruction directly following the jump instruction. Starting with 3.12, some jump instructions can be followed by a :opcode:`CACHE` instruction. When the argument of such an instruction is 0, it will jump to the first non-``CACHE`` instruction following the jump instruction. 

As a consequence, the presence of the :opcode:`CACHE` instructions is transparent for forward jumps but need to be taken into account when reasoning about backward jumps.

Reply:
I think I see what you mean now. 

Can we just say that "the arg of a jump is the offset from the instruction that appears immediately after the JUMP instruction's CACHE entries" ?

Reply:
Can we agree on the following ? I would prefer to insist on the forward backward asymmetry

The argument of a jump is the offset of the target relative to the instruction that appears immediately after the jump instruction's CACHE entries. 

As a consequence, the presence of the CACHE instructions is transparent for forward jumps but need to be taken into account when reasoning about backward jumps.

Reply:
Sure. I think "need --> needs" though.

Reply:
Sure